Assets/Scripts/InputManager/InputManager.cs

```diff
@@ -2,31 +2,64 @@
 using Common;
 using UnityEngine;
 using BTManager;
+using Player;
+using StateMachine;
+using UnityEngine.SceneManagement;
+using Replay;
 
 namespace Inputs
 {
     public class InputManager : Singleton<InputManager>
     {
         private List<InputComponent> inputComponentList = new List<InputComponent>();
 
-        private bool onPaused = false;
+        protected override void Awake()
+        {
+            base.Awake();
+        }
 
         private void Start()
         {
-            GameManager.Instance.GamePaused += OnPaused;
-            GameManager.Instance.GameUnpaused += OnUnPaused;
+            GameManager.Instance.GameStarted += OnGameStart;
         }
 
         // Update is called once per frame
         void Update()
         {
-            if (onPaused == false)
+            if (GameManager.Instance.currentState.gameStateType == GameStateType.Game)
```

Comment:
duplication - if else is not needed. when replay mode is entered - replay manager should feed the data to input manager